lib/Launcher.js

```diff
@@ -142,20 +142,12 @@ class Launcher {
      */
     function killChrome() {
       helper.removeEventListeners(listeners);
-      if (chromeProcess.pid) {
-        if (temporaryUserDataDir) {
-          // Force kill chrome.
-          if (process.platform === 'win32')
-            childProcess.execSync(`taskkill /pid ${chromeProcess.pid} /T /F`);
-          else
-            process.kill(-chromeProcess.pid, 'SIGKILL');
-        } else {
-          // Terminate chrome gracefully.
-          if (process.platform === 'win32')
-            childProcess.execSync(`taskkill /pid ${chromeProcess.pid}`);
-          else
-            process.kill(-chromeProcess.pid, 'SIGTERM');
-        }
+      if (chromeProcess.pid && temporaryUserDataDir) {
+        // Force kill chrome.
+        if (process.platform === 'win32')
+          childProcess.execSync(`taskkill /pid ${chromeProcess.pid} /T /F`);
```

Comment:
this is racing with protocol's browser.close. What happens if browser has already been closed? Could it be that we might terminate something else (in case this pid got reused)?

Reply:
Outdated, I check !chromeProcess.killed && chromeProcess.connected

Reply:
This does reduce probability but doesn't fix the race.